README.md

```diff
@@ -121,22 +121,22 @@ Sticky React Table supports a host of properties which allow you to completely c
-| infiniteScrollCellRenderer   | `Node` or `Function` | `undefined` | Custom content to dispaly within cells of additional loader rows.               | No       |
+| Property                     | Type                 | Default     | Description                                                                                         | Required |
+| ---------------------------- | ---------------------| ----------- | --------------------------------------------------------------------------------------------------- | -------- |
+| data                         | `Array`              | `undefined` | The array of data to be used by the table to render rows and cells.                                 | **Yes**  |
+| children                     | `Column`             | `undefined` | The `Column` component. It does not accept any other children.                                      | **Yes**  |
+| fixed                        | `Number`             | `undefined` | The number of fixed columns in the table.                                                           | No       |
+| rowSelection                 | `Bool`               | `true`      | This property determines whether checkbox column is rendered.                                       | No       |
+| idKey                        | `String`             | `id`        | A key used to uniquely identify data.                                                               | No       |
+| rowClassName                 | `String`             | `undefined` | Custom classes for a row.                                                                           | No       |
+| headerClassName              | `String`             | `undefined` | Custom classes for the header row.                                                                  | No       |
+| selectedRows                 | `Array`              | `undefined` | An array of ids to make the table act as a controlled selection component.                          | No       |
+| infiniteScrollTotalCount     | `Number`             | `undefined` | Total number of rows which can be loaded using infinite loader.                                     | No       |
+| infiniteScrollLoadMore       | `Function`           | `undefined` | Invoked when a new page is requested when using infinite loader.                                    | No       |
+| infiniteScrollThreshold      | `Number`             | `10`        | The number of rows before the end of the table that will trigger a call to infiniteScrollThreshold. | No       |
+| infiniteScrollLoaderRowCount | `Number`             | `5`         | The number of additional rows to display at the bottom for progress indication.                     | No       |
+| infiniteScrollPageSize       | `Number`             | `30`        | A number of rows loaded with each portion.                                                          | No       |
```

Comment:
Is this an internal prop? What is the use of this and why is this required for the table? Afaik, table shouldn't really need to know how much we want to load. Just take in the new data, and render it. If the keys provided are correct, it won't rerender all the rows.

Reply:
No it's not an internal one. it's needed to determine the next page number to load which in turn is passed to `infiniteScrollLoadMore` call. It's also used internally to determine which "page" we're currently loading.

Reply:
I think in this, we might be doing a lot more than what we should, within the scope of the table. The fact that we natively support infinite loading is a feature enough. I think all of this should be handled by the user, and not by us. 

I genuinely believe we are overcomplicating a rather straightforward feature. 

What are your thoughts regarding the same?

Reply:
I think it's still a very basic implementation. This is how many infinite loaders work taking either page number or number of rows to load (which is in some sense the same stuff) and doing calculations based on it. 

Again we need to track current page inside the component for a few reasons:
- to detect that a certain page is loaded - to avoid multiple requests of the same page
- and what's really important for scroll performance - to do less calculations in `onScroll` handler: https://github.com/samrith-s/sticky-react-table/blob/a2b4dab62667b211fa7f93408ebc1cfdb3e4f553/src/components/Table.js#L417

Take a look and react-virtualized `InfiniteLoader. minimumBatchSize` prop - it's the same stuff.

Reply:
Hey,

I was looking on it before your comment. What we can do is refactor this a bit to reduce the props or rename them a bit better in future versions. For now, let's go with this.

Reply:
@samrith-s  I honestly don't think this is the final implementation ) So we'll have bunch of time to think how to improve it depending on use cases :)